### PR TITLE
Improved report library

### PR DESCRIPTION
**Pull request by sidezrw**
## Proposed changes

- Improves modularization of the report library. These are initial changes to provide support for a report visualization tool in the future.
- Adds per benchmark and global report time unit selection using config files.

## Types of changes

What types of changes does your code introduce to the HEBench Frontend?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/hebench/frontend/blob/main/CONTRIBUTING.md) doc
- [x] Current formatting and unit tests / base functionality passes locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: common-lib/modules/general/src/hebench_utilities.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    std::size_t pos_e = retval.find_last_of("eE");
+    if (pos_e != std::string::npos)
+    {
+        double coeff;
+        std::string s_sci_part = retval.substr(pos_e);
+        std::string s_coeff    = retval.substr(0, pos_e);
+        ss                     = std::stringstream(s_coeff);
+        ss >> coeff;
+        s_coeff.erase(s_coeff.find_last_not_of("0") + 1);
+        //        if (!s_coeff.empty() && s_coeff.back() == '.')
```
Comment (jobottle) [inline common-lib/modules/general/src/hebench_utilities.cpp:70]: Looks like dead/commented code. Remove / add comment stating why it is being left in.

**Review by jobottle — APPROVED** [review]
LGTM

File: common-lib/modules/general/src/hebench_utilities.cpp (hunk 1 above)
Comment (sidezrw) [reply common-lib/modules/general/src/hebench_utilities.cpp]: Removed/documented!

**Review by kylanerace — APPROVED** [review]
new changes lgtm

**Pull request merged**